### ISSUE TITLE
mqtt-streaming ControlPacketFlags.QoS* has incorrect meaning when subscribing

### ISSUE THREAD

**Issue by ohze** [akka/alpakka#2731]
The following commands:
+ `Command(Subscribe(Seq("topic" -> QoSAtLeastOnceDelivery))`  will subscribe with qos 2
+ `Command(Subscribe(Seq("topic" -> QoSExactlyOnceDelivery))`  will be encoded to an invalid mqtt package (so, mqtt server will disconnect the subscribing client)

### Versions used 

akka-stream-alpakka-mqtt-streaming 3.0.3  (akka 2.6.14)

### Expected Behavior
correct meaning

### Actual Behavior
incorrect meaning

### Relevant logs
```
Test for qos ControlPacketFlags(0)
SubAck(PacketId(1),Vector(ControlPacketFlags(0)))
Done 0

Test for qos ControlPacketFlags(2)
SubAck(PacketId(1),Vector(ControlPacketFlags(2)))
Done 2

Test for qos ControlPacketFlags(4)
java.lang.AssertionError: assertion failed
  scala.Predef$.assert(Predef.scala:264)
  ammonite.$file.s$.$anonfun$res_6$1(s.sc:26)
  ammonite.$file.s$.$anonfun$res_6$1$adapted(s.sc:9)
  scala.collection.immutable.List.foreach(List.scala:333)
  ammonite.$file.s$.<clinit>(s.sc:9)
```
mosquitto log:
```
1631793666: mosquitto version 2.0.12 starting
1631793666: Config loaded from /mosquitto/config/mosquitto.conf.
1631793666: Opening ipv4 listen socket on port 1883.
1631793666: Opening ipv6 listen socket on port 1883.
1631793666: mosquitto version 2.0.12 running
1631793674: New connection from 172.17.0.1:59778 on port 1883.
1631793674: New client connected from 172.17.0.1:59778 as 0 (p2, c1, k60).
1631793674: New connection from 172.17.0.1:59782 on port 1883.
1631793674: Client 0 closed its connection.
1631793674: New client connected from 172.17.0.1:59782 as 2 (p2, c1, k60).
1631793674: Client 2 closed its connection.
1631793674: New connection from 172.17.0.1:59786 on port 1883.
1631793674: New client connected from 172.17.0.1:59786 as 4 (p2, c1, k60).
1631793674: Invalid QoS in subscription command from 172.17.0.1, disconnecting.
1631793674: Client 4 disconnected due to malformed packet.
```

### Reproducible Test Case
test.sc
```scala
import $ivy.`com.lightbend.akka::akka-stream-alpakka-mqtt-streaming:3.0.3`
import scala.concurrent._, duration._
import akka._, actor._, util._, stream._, stream.scaladsl._
import akka.stream.alpakka.mqtt._, streaming._, streaming.scaladsl._
import ControlPacketFlags.{QoSAtMostOnceDelivery, QoSAtLeastOnceDelivery, QoSExactlyOnceDelivery}

implicit val system: ActorSystem = ActorSystem("MqttStreaming")

Seq(QoSAtMostOnceDelivery, QoSAtLeastOnceDelivery, QoSExactlyOnceDelivery).foreach { qos =>
  println(s"\nTest for qos $qos")
  val id = qos.underlying.toString

  val settings = MqttSessionSettings()
  val session = ActorMqttClientSession(settings)
  val conn = Tcp().outgoingConnection("localhost", 1883)
  val mqttFlow = Mqtt.clientSessionFlow(session, ByteString(id)).join(conn)
  val (commands, events) = Source.queue(10).via(mqttFlow).preMaterialize()

  commands.offer(Command(Connect(id, ConnectFlags.CleanSession)))
  commands.offer(Command(Subscribe(Seq(s"topic$id" -> qos))))

  val future = events.
    collect { case Right(Event(s: SubAck, _)) => println(s); s }.
    runWith(Sink.headOption)
  val maybeAck = Await.result(future, 20.seconds)
  assert(maybeAck.nonEmpty)
  println(s"Done $id")
}
```

```bash
echo "listener 1883" >> m.conf
echo "allow_anonymous true" >> m.conf

docker run --name mqtt --rm -a stdout -a stderr -p 1883:1883 -v `pwd`/m.conf:/mosquitto/config/mosquitto.conf eclipse-mosquitto:2.0.12

# require: http://ammonite.io/#Ammonite-REPL
amm test.sc
```

**Comment by ennru** [comment]
@ohze Did you have a look at the implementation to suggest a fix?

**Comment by sbmpost** [comment]
@ennru I can confirm this is broken.

The encoding is done incorrectly at https://github.com/akka/alpakka/blob/3d8b41ffe8c5904b8d2d197d642939d26a84d4ee/mqtt-streaming/src/main/scala/akka/stream/alpakka/mqtt/streaming/model.scala#L701

Note that the Publish packet uses bits 2 and 1 to encode the QoS. So in that case the encoding is OK.
The Subscribe packet however uses bits 1 and 0 to encode the QoS. In that case the encoding is NOT OK (the flag needs to be shifted one position to the right).

References:
http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718038
http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718063

**Comment by ennru** [comment]
Thank you for digging into this. We'll try to get it fixed before the upcoming Alpakka release.

**Comment by ennru** [comment]
A separation of the constants for Publish and Subscribe is now merged with #2969.
To be released very soon.
